core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/app/AppContext.java

```diff
@@ -100,4 +100,15 @@
     String getProjectsRoot();
 
     void setProjectsRoot(String projectsRoot);
+
+    /**
+     * Sets URL to send requests to workspace agent.
+     *
+     * @param wsAgentURL
+     *         workspace agent url
+     */
+    void setWsAgentURL(String wsAgentURL);
```

Comment:
Dont provide setters in context objects, set this information in context implementation, because every user may set own ws agent url, that is not good